sources/packages/backend/libs/services/src/student-loan-balance/student-loan-balance-shared.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { StudentLoanBalance } from "@sims/sims-db";
+import { Repository } from "typeorm";
+@Injectable()
+export class StudentLoanBalanceSharedService {
+  constructor(
+    @InjectRepository(StudentLoanBalance)
+    private readonly studentLoanBalanceRepo: Repository<StudentLoanBalance>,
+  ) {}
+
+  async getLatestCSLPBalance(studentId: number): Promise<StudentLoanBalance[]> {
+    return this.studentLoanBalanceRepo.find({
```

Comment:
`Limit` can be used to return just 1 record.

Reply:
You mean take? if there are no records for student in the studentLoanBalance table, the query will fail. So did not included it here.

Reply:
The below query works even in an empty table. Please adjust the query to return only the most recent record.
```sql
select * from sims.student_loan_balances limit 1
```

Reply:
Changed the typeorm to querybuilder to accommodate the Limit.